modules/components/src/DropDown/DropDown.js

```diff
@@ -5,135 +5,160 @@ import { css } from '@emotion/react';
+					selectedItem,
+					inputValue,
+					highlightedIndex,
+				}) => (
+					<div className="dropDownHeader">
+						<button
+							aria-label={`Show columns to select`}
+							className="dropDownButton"
+							{...getButtonProps({
+								disabled: disableDownloads,
+								onClick: this.handleToggleMenu,
+							})}
+						>
+							<div className="dropDownButtonContent">{children}</div>
+							<ArrowIcon
+								css={css`
+									margin-left: 0.3rem;
+									margin-top: 0.1rem;
+								`}
+								fill={customArrowColor || themeArrowColor}
+								pointUp={isOpen}
+								transition={customArrowTransition || themeArrowTransition}
+								{...themeArrowProps}
+							/>
+						</button>
 
-            {isOpen && (
-              <div
-                className={`dropDownContent ${singleSelect ? 'single' : 'multiple'}`}
-                style={{
-                  right: align === 'right' ? 0 : 'auto',
-                  left: align === 'right' ? 'auto' : 0,
-                }}
-                {...(singleSelect && { onClick: this.handleToggleMenu })}
-              >
-                {items.map((item, index) => {
-                  const { id, ...itemProps } = getItemProps({
-                    item,
-                    index,
-                    disabled: item.exporterRequiresRowSelection && !hasSelectedRows,
-                  });
-                  const label = itemToString(item);
-                  const labelIsComponent = React.isValidElement(label);
-                  return (
-                    <div
-                      className={`dropDownContentElement${
-                        labelIsComponent ? ' custom' : ' clickable'
-                      }`}
-                      key={item.id || id}
-                      {...itemProps}
-                    >
-                      {label}
-                      {!(singleSelect || labelIsComponent) && (
-                        <input
-                          readOnly
-                          type="checkbox"
-                          checked={selectedItem.indexOf(item) > -1}
-                          aria-label={`Select column ${
-                            item.id || (typeof label === 'string' ? label : id)
-                          }`}
-                        />
-                      )}
-                    </div>
-                  );
-                })}
-              </div>
-            )}
-          </div>
-        )}
-      </Downshift>
-    );
-  }
+						{isOpen && (
+							<div
+								className={`dropDownContent ${singleSelect ? 'single' : 'multiple'}`}
+								style={{
+									right: align === 'right' ? 0 : 'auto',
+									left: align === 'right' ? 'auto' : 0,
+								}}
+								{...(singleSelect && { onClick: this.handleToggleMenu })}
+							>
+								{enableSelectColumnDropdownTextFilter && (
+									<TextFilter
+										aria-label={`Search data`}
+										onChange={({ value }) => this.setState({ searchText: value || '' })}
```

Comment:
Think I'd abstract this function out of the return, so a new function is not instantiated with every render.

e.g. `onChange={handleFilter}`

Reply:
thanks for the suggestion! code refactored as suggested.